server/src/infra/repositories/metadata.repository.ts

```diff
@@ -20,31 +22,45 @@ import {
 import { ImmichLogger } from '@app/infra/logger';
 import { Inject } from '@nestjs/common';
 import { InjectDataSource, InjectRepository } from '@nestjs/typeorm';
-import { DefaultReadTaskOptions, exiftool, Tags } from 'exiftool-vendored';
+import { DefaultExiftoolArgs, DefaultReadTaskOptions, ExifTool, Tags } from 'exiftool-vendored';
 import * as geotz from 'geo-tz';
 import { getName } from 'i18n-iso-countries';
 import { createReadStream, existsSync } from 'node:fs';
 import { readFile } from 'node:fs/promises';
 import * as readLine from 'node:readline';
+import { Subscription } from 'rxjs';
 import { DataSource, DeepPartial, QueryRunner, Repository } from 'typeorm';
 import { DummyValue, GenerateSql } from '../infra.util';
 
 type GeoEntity = GeodataPlacesEntity | GeodataAdmin1Entity | GeodataAdmin2Entity;
 type GeoEntityClass = typeof GeodataPlacesEntity | typeof GeodataAdmin1Entity | typeof GeodataAdmin2Entity;
 
 export class MetadataRepository implements IMetadataRepository {
+  private configCore: SystemConfigCore;
+  private subscription: Subscription | null = null;
+
   constructor(
     @InjectRepository(ExifEntity) private exifRepository: Repository<ExifEntity>,
     @InjectRepository(GeodataPlacesEntity) private readonly geodataPlacesRepository: Repository<GeodataPlacesEntity>,
     @InjectRepository(GeodataAdmin1Entity) private readonly geodataAdmin1Repository: Repository<GeodataAdmin1Entity>,
     @InjectRepository(GeodataAdmin2Entity) private readonly geodataAdmin2Repository: Repository<GeodataAdmin2Entity>,
+    @Inject(ISystemConfigRepository) configRepository: ISystemConfigRepository,
     @Inject(ISystemMetadataRepository) private readonly systemMetadataRepository: ISystemMetadataRepository,
     @InjectDataSource() private dataSource: DataSource,
-  ) {}
+  ) {
+    this.configCore = SystemConfigCore.create(configRepository);
+  }
 
   private logger = new ImmichLogger(MetadataRepository.name);
+  private exiftool: ExifTool;
 
   async init(): Promise<void> {
+    if (!this.subscription) {
+      this.subscription = this.configCore.config$.subscribe(() => this.init());
```

Comment:
This creates a new exiftool instance with the updated setting, but it wouldn't close the previous one.